app/lib/actions/katello/capsule_content/sync.rb

```diff
@@ -54,6 +59,70 @@ def get_repository_ids(capsule, environment, content_view, repository)
           repository_ids
         end
 
+        def repos_needing_updates(capsule_content, environment, content_view)
+          need_importer_update = repos_needing_importer_updates(capsule_content, environment, content_view)
+          need_distributor_update = repos_needing_distributor_updates(capsule_content, environment, content_view)
+          (need_distributor_update && need_importer_update).uniq
+        end
+
+        def repos_needing_distributor_updates(capsule, environment, content_view)
+          repos = capsule.repos_available_to_capsule(environment, content_view)
+          update_distributors = repos.select do |repo|
+            generated_distributors = repo.generate_distributors
+            repo_details = capsule.pulp_repo_facts(repo.pulp_id)
+            next unless repo_details
+            capsule_distributors = repo_details["distributors"]
+            distributors_changed?(capsule_distributors, generated_distributors)
+          end
+          update_distributors
```

Comment:
Spit balling on this one as an example of how you might think about it, if it doesn't make sense to attach to the repo object itself, at least take a look at the logic updates to see if they match expectation:


```
       def repos_needing_distributor_updates(capsule, environment, content_view)
          repos = capsule.repos_available_to_capsule(environment, content_view)
          repos.select do |repo|
            repo_details = capsule.pulp_repo_facts(repo.pulp_id)
            next unless repo_details          
            capsule_distributors = repo_details["distributors"]
            repo.distributors_changed?(repo_details["distributors"])
          end
       end

in glue/pulp/repo.rb:

      def distributors_changed?(distributors)
           generated_distributors = repo.generate_distributors.map { |dist| object_to_hash(dist) }
           distributors.each do |dist|
             dist.merge!(dist["config"])
             dist.delete("config")
           end
 
           generated_distributors.any? do |gen_dist|
             capsule_distributors.any? do |cap_dist|
               (gen_dist.to_a - cap_dist.to_a).empty?
             end
           end
         end  
    end
```